pkg/registry/resource/resourceclaimtemplate/strategy_test.go

```diff
@@ -815,11 +825,15 @@ func TestStrategyUpdate(t *testing.T) {
 			newObj.ResourceVersion = "4"
 
 			strategy.PrepareForUpdate(ctx, newObj, oldObj)
+			expectedErrLen := len(tc.expectValidationErrors)
 			if errs := strategy.ValidateUpdate(ctx, newObj, oldObj); len(errs) != 0 {
-				assert.ErrorContains(t, errs[0], tc.expectValidationError, "the error message should have contained the expected error message")
+				assert.Len(t, errs, expectedErrLen, "exact number of errors expected")
+				for i, expectErr := range tc.expectValidationErrors {
+					assert.ErrorContains(t, errs[i], expectErr, "the error message should have contained the expected error message")
+				}
```

Comment:
```suggestion
				if assert.Len(t, errs, expectedErrLen, "exact number of errors expected") {
					for i, expectErr := range tc.expectValidationErrors {
						assert.ErrorContains(t, errs[i], expectErr, "the error message should have contained the expected error message")
					}
				}
```

Don't compare the elements unless the length is the same, it would panic the test.

Reply:
/lgtm
/hold

Feel free to lift the hold. A panic is still going to show up as a test failures, just not as a nice one.